File: 338-counting-bits/338-counting-bits.py
class Solution(object):
    def countBits(self, n):
        """
        :type n: int
        :rtype: List[int]
        """
        result = [0]
        
        if n == 0:
            return result
        else: 
            for num in range(1, n + 1):
                binString = (str(bin(num))[2::])
                count = 0
                for i in binString:
                    if i == "1":
                        count += 1
                result.append(count)
        return result
            

File: 338-counting-bits/338-counting-bits.java
class Solution {
    public int[] countBits(int n) {
        int[] arr = new int[n + 1];
        
        for (int i = 0; i <= n; i++) {
            arr[i] = hammingWeight(i);
        }
        
        return arr;
    }
    
    public int hammingWeight(int n) {
        int counter = 0;
        while (n != 0) {
            if ((n & 1) == 1) {
                counter++;
            }
            
            n = n >>> 1; 
        }
        
        return counter;
    }
}